lib/Interpreter/CIFactory.cpp

```diff
@@ -893,16 +902,64 @@ namespace {
     // of the two is the underlying compiler.
 
 #ifdef __clang__
-    PPOpts.addMacroDef("__CLING__clang__=" ClingStringify(__clang__));
+  #if defined(__apple_build_version__)
+     // __apple_build_version__ >= 8000031 Xcode 8
+     #if __apple_build_version__ >= 7030029
```

Comment:
The changes to the macros are here because the commit for falling back if a deprecated method is removed wouldn't apply cleanly to master without that commit which I authored because I needed the info.  If you don't want it thats fine.